### DISCUSSION TITLE
completely out of my depth with electronics

### DISCUSSION THREAD

**Discussion by Scientartistroaster** [artisan-roaster-scope/artisan#1641] [Q&A]
*UPDATED* 

Hello Everyone. I am trying to figure out if we would be able to use artisan with our current roaster and am finding that everything comes to be greek to me. please forgive me if I missed a post that might've helped me with my specific situation, so far I have not found any.

I am currently roasting on a custom built roaster that was made in country(South Africa) the roaster name is "Blazing Flame". it was made by mechanical concepts( with whom I have not been able to get in contact with) and sold by singingbeanscoffee.co.za. She currently uses Delta computing and control hardware which includes: an lcd screen, Delta ME300 controller drum speed, DVP-14SS2 & DVP-02TU( I believe to do with temperature reading). 

I have contacted some of the other people that use the same roaster and so far none are using any software to track roasts beyond the machines provided hardware. Unfortunately due to the hands on approach to the roaster I can not take adequate handwriten notes of temp changes during roasts, which is why I am hoping to be able to figure out how to hook her up to artisan.

I am attaching some pictures of the components
![IMG_20240729_103510](https://github.com/user-attachments/assets/df33fcd2-86ab-4287-b991-74022a68f98a)
![IMG_20240729_103533](https://github.com/user-attachments/assets/97d8bb8d-37a8-445d-947f-90c0f549f214)
![IMG_20240729_113737](https://github.com/user-attachments/assets/63a32dc3-6b62-49eb-b961-45382f38128b)
 
*Update*
![IMG_20240122_080051](https://github.com/user-attachments/assets/5db544d0-8085-4a8b-b394-d7e1fd8419a1)
![IMG_20240122_080323](https://github.com/user-attachments/assets/a7a5e6bb-64fd-40c4-a4cf-37cc8ebd3425)
![IMG_20240122_080329](https://github.com/user-attachments/assets/44b4c2a2-0759-4a1d-9a73-c3f98b8e52f4)
![IMG_20240122_080341](https://github.com/user-attachments/assets/c20a2947-6ae3-4b39-9076-15fc06201241)

Found these from previous correspondence I had with someone from another company. will add one of the activated screen as she currently is tomorrow.

**Comment by poundy** [comment]
First up, I would try to get an electrical wiring diagram / picture of everything at a wider scale than you've got right now - for example I can see a translucent blue USB cable there in a port marked "host" plugged into the display that would be interesting to know what it's connected to, and there's a serial link in COM2 to track.

The other thing you haven't captured is the "human" side. Is there any other visible temperature displays other than on the LCD panel? 

Usually these kinds of questions end up with the unfortunate answer that unless the manufacturer is forthcoming in sponsoring the Artisan project to integrate their equipment into Artisan, there's little that can be done for you in true integration. And if there's no documentation available, any attempts to "reverse engineer" the roaster is also up to the only person who has one in their posession, as it's likely a lot of trial and error and you need access to the equipment to do that. At a minimum, Artisan would ordinarily be connected to whatever control and measurement equipment the Delta display (https://industrialautomation.delta-emea.com/en/dop-107ev-1810.htm) is reading from and writing to - likely over the USB and serial connections I mentioned above.

The alternative here is that you simply "append" your own measurement equipment to the roaster, that is then connected to Artisan. Usually that means buying a Phidget and temperature sensors, and connecting that to a computer. https://artisan-scope.org/devices/phidgets/ can help you start to see the variety of options.  It completely depends on where you want to stop - there's a lot that could be done but you end up needing to know more the deeper you go into this.

**Reply by poundy** [reply]
I'd also say that the path to quickest success is to work with the manufacturer or designer. If you can get the details of what is talking to what, and how, then there's a chance that you can redo that in Artisan. Fundamenally though, the manufacturer needed to have designed the roaster equipment to be "open" to this additional connection, and it'll be hard to do anything more than do a separate measurement system if they are not forthcoming

**Reply by Scientartistroaster** [reply]
Thanks for the input @poundy. I will take a few more pictures and include some of the overall setup of things, hopefully by end of day tomorrow.

With regards to the control and measuring equipment, as far as I can tell the DVP's(seen in picture no'3) read and control the following respectivly: 
Temperature(https://industrialautomation.delta-emea.com/en/dvp-ss2-series-1179.htm)

Speed(https://filecenter.deltaww.com/Products/download/06/060301/Manual/DELTA_IA-PLC_DVP02TU-S_DVP02TK-S_OM_EN_20230425.pdf) 

I will find out from the people who sold it if they might be able to help with a diagram or some more information but don't really have high hopes.
